File: CSharpPools/PoolsLibrary/Controller/InvalidPoolOperationException.cs
namespace PoolsLibrary.Controller
{
    class InvalidPoolOperationException<TK,TV> : PoolException<TK>
    {
        public TV Object { get; private set; }

        public InvalidPoolOperationException(TV operationObject = default(TV),
                                             TK key = default(TK),
                                             string message = null)
            : base(key, message)
        {
            Object = operationObject;
        }

        public InvalidPoolOperationException()
        {
        }
    }
}

File: JavaPools/PoolsLibrary/src/main/java/ru/urfu/javapools/poolslibrary/controller/InvalidPoolOperationException.java
package ru.urfu.javapools.poolslibrary.controller;

@SuppressWarnings("serial")
public class InvalidPoolOperationException extends PoolException {

	private Object _object;
	
	public InvalidPoolOperationException (Object key, Object object, String message) {
		super(key,message);
		_object = object;
	}
	
	public Object getObject() {
		return _object;
	}
}
